sycl/include/CL/sycl/handler.hpp

```diff
@@ -522,6 +522,79 @@ class __SYCL_EXPORT handler {
     return true;
   }
 
+#ifndef __SYCL_DEVICE_ONLY__
+  /// Copies the content of memory object accessed by Src into the memory
+  /// pointed by Dst.
+  ///
+  /// \param Src is a source SYCL accessor.
+  /// \param Dst is a pointer to destination memory.
+  template <typename TSrc, typename TDst, int Dim, access::mode AccMode,
+            access::target AccTarget, access::placeholder IsPH>
+  detail::enable_if_t<(Dim > 0)>
+  copyAccToPtrHost(accessor<TSrc, Dim, AccMode, AccTarget, IsPH> Src,
+                   TDst *Dst) {
+    range<Dim> Range = Src.get_range();
+    parallel_for<class __copyAcc2Ptr<TSrc, TDst, Dim, AccMode, AccTarget, IsPH>>
+        (Range, [=](id<Dim> Index) {
+      size_t LinearIndex = Index[0];
+      for (int I = 1; I < Dim; ++I)
+        LinearIndex += Range[I] * Index[I];
+      ((TSrc *)Dst)[LinearIndex] = Src[Index];
```

Comment:
Can we use C++ casts here?

Reply:
If you don't mind, I'll create another '[SYCL][NFC] Fix c-style casts in handler::copy()' pull-request, and in parallel with that will upload USM support for reduction, after unblocking it by merging 'this' PR.